pkg/provisioner/ironic/ironic.go

```diff
@@ -192,15 +196,17 @@ func newProvisionerWithIronicClients(host metal3v1alpha1.BareMetalHost, bmcCreds
 	// we need.
 	clientIronic.Microversion = "1.56"
 	p := &ironicProvisioner{
-		host:      host,
-		status:    &(host.Status.Provisioning),
-		bmcAccess: bmcAccess,
-		bmcCreds:  bmcCreds,
-		client:    clientIronic,
-		inspector: clientInspector,
-		log:       log.WithValues("host", host.Name),
-		publisher: publisher,
-	}
+		host:               host,
+		status:             &(host.Status.Provisioning),
+		bmcAccess:          bmcAccess,
+		bmcCreds:           bmcCreds,
+		client:             clientIronic,
+		inspector:          clientInspector,
+		publisher:          publisher,
+		watchAllNamespaces: os.Getenv("WATCH_NAMESPACE") == "",
+		podNamespace:       os.Getenv("POD_NAMESPACE"),
```

Comment:
Is it possible to pass these two params along with the other params in the factory method? Otherwise this strictly coupling could affect also the unit testing.